mm2src/coins/lp_coins.rs

```diff
@@ -2962,8 +3005,24 @@ pub fn address_by_coin_conf_and_pubkey_str(
                 _ => ERR!("Platform protocol {:?} is not BCH", platform_protocol),
             }
         },
-        CoinProtocol::TENDERMINT { .. } => {
-            ERR!("address_by_coin_conf_and_pubkey_str is not implemented for TENDERMINT protocol yet!")
+        CoinProtocol::TENDERMINT(protocol) => tendermint::account_id_from_pubkey_hex(&protocol.account_prefix, pubkey)
+            .map(|id| id.to_string())
+            .map_err(|e| e.to_string()),
+        CoinProtocol::TENDERMINTTOKEN(proto) => {
+            let platform_conf = coin_conf(ctx, &proto.platform);
+            if platform_conf.is_null() {
+                return ERR!("platform {} conf is null", proto.platform);
+            }
+            // TODO is there any way to make it better without duplicating the prefix in the IBC conf?
```

Comment:
Isn't this implemented approach good enough?

Reply:
@sergeyboyko0791 I have left this TODO. I see this as a bit too much of boilerplate code, but I don't have an idea for now on how to make it better.